src/Summoner/Question.hs

```diff
@@ -45,13 +46,17 @@ printQuestion question (def:rest) = do
     putTextLn $ "/" <> restSlash
 printQuestion question [] = T.putStrLn question
 
-choose :: (Show a, IsString a) => Text -> [a] -> IO a
+choose :: forall a. (Show a, Read a) => Text -> [a] -> IO a
 choose question choices = do
     let showChoices = map show choices
     printQuestion question showChoices
     answer <- prompt
     if | T.null answer -> pure (Unsafe.head choices)
-       | answer `elem` showChoices -> pure $ fromString $ T.unpack answer
+       | answer `elem` showChoices -> do
+           let mAnswer = readMaybe @a $ T.unpack answer
+           case mAnswer of
+               Nothing -> error "Unreadable answer type"
+               Just a  -> pure a
```

Comment:
Is it possible to make it better? Without `Nothing`?

Reply:
Well, I meant a little bit different think.
I want to get rid of `showChoices`. It's not needed at all to translate all to text and then back. What I propose if to do something like
```haskell
if T.null answer 
    then pure (Unsafe.head choices)
    else case readMaybe @a answer of
            Just a  -> pure a	       
            Nothing -> do
               errorMessage "This wasn't a valid choice."
               choose question choices
```